server/src/main/java/org/elasticsearch/index/mapper/ScriptParams.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.mapper;
+
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.script.ScriptService;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+public final class ScriptParams {
```

Comment:
I think params are intended for something else in the context of scripts. As far as I understand this class holds the original script as well as the script in its compiled form, as well as the param needed to parse it from the mappings.

Reply:
Params here are mapper params - we need to hold the original Script for serialization and comparison during merges, but we also only want to compile the script once.

Reply:
The name of the class is confusing because `Script` has a `getParams` method. This sort of parameter is going to be super common - should it just be a static method inside `FieldMapper.Parameter`? You've done that a few times but not always. I know you had a good reason when you made `TextParams` but I sure don't remember it.

Reply:
Now that serialization and equality checking is handled directly in MapperScript I think I can just move this parameter definition into FieldMapper.